lib/ClusterProvider.py

```diff
@@ -0,0 +1,57 @@
+import common
+import subprocess
+import os
+import configs
+from robot.api import logger
+
+ROBOT_LIBRARY_SCOPE = 'SUITE'
+
+
+def load_script(provider):
+    path = 'scripts/cluster_providers/'
+    if provider == 'kube':
+        return path + 'kube.sh'
+    if provider == 'kind':
+        return path + 'kind.sh'
+    return "invalid"
+
+
+def setup_cluster(provider, metadata):
+    return SetupCluster(provider, metadata)
+
+
+class SetupCluster():
+    def __init__(self, provider, metadata):
+        self.provider = provider
+        self.metadata = metadata
+        self.base_cmd = 'bash -c'
+        self.cluster_name = 'helm-acceptance-test-' + self.metadata
+
+    def setup_cluster(self):
+        self.call_cluster_provisioner_script('create_cluster')
+        configs.AUTH_COMMAND = self.call_cluster_provisioner_script(
+            'get_cluster_auth')
+
+    def delete_cluster(self):
+        self.call_cluster_provisioner_script('delete_cluster')
+
+    def get_cluster_auth(self):
+        configs.AUTH_COMMAND = self.call_cluster_provisioner_script(
```

Comment:
auth is already set in the cofigs we are redoing it I believe.

Reply:
yea this looks unnecessary, will remove it